scripts/fix_security_group.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import gzip
+import json
+import logging
+import os
+import platform
+import urllib.parse
+import urllib.request
+import urllib.response
+
+
+def get_http_response_str(response: urllib.response.addinfourl) -> str:
+    if response.info().get("Content-Encoding") == "gzip":
+        pagedata = gzip.decompress(response.read())
+    elif response.info().get("Content-Encoding") == "deflate":
+        pagedata = response.read()
+    else:
+        pagedata = response.read()
+
+    return pagedata
+
+
+def get_bearer_token(
+    refresh_token: str, cloud_base_domain: str = "cloud.ibm.com"
+) -> str:
+    url = f"https://iam.{cloud_base_domain}/identity/token"
```

Comment:
This is going to break for customers going through the CSE endpoints or IAM VPE. The domain is private.iam.cloud.ibm.com in this case.

Reply:
Are the private endpoints for IAM and endpoints for IaaS resources (your other comment) related? If we have a single boolean that can be supplied to "use_private_endpoints" cover both cases?

Reply:
Latest version now has support for private endpoints, enabled by setting new python parameter to "true" (default of "false"). Intent is for that input variable to come from the terraform call.